### PR TITLE
Feature/datacloud terms and conditions

### PR DESCRIPTION
**Pull request by jakubneubauer**
When a "DataCloud Sign-in dialog" is shown and after filling credentials, and if T&C have not yet been accepted (e.g. during registration), this step appears and user has a chance to accept T&C of DataCloud services:
screenshot from 2016-06-01 13 43 25

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java [hunk 1]
```diff
@@ -334,26 +400,68 @@ public void run() {
...
                 invalidCredentialsLabel.setText("&nbsp;");
                 logger.debug("Sign in to DataCloud succeeded. User name: {}", userName);
 
-                RemoteServersUtils.addRemoteServer(_configuration.getEnvironment(), RemoteDescriptorProvider.DATACLOUD_SERVER_NAME, RemoteDescriptorProvider.DATACLOUD_URL, userName, pass);
-
-                CardLayout cardLayout = (CardLayout) cards.getLayout();
-                cardLayout.show(cards, SUCCESS_CARD);
-                dontShowAgainCheckBox.setVisible(false);
+                addRemoteServer();
+                //dontShowAgainCheckBox.setVisible(false);
```
Comment (rposkocil) [inline desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java]: Would you straight delete it? If it isn't needed.

File: desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java [hunk 2]
```diff
@@ -334,26 +400,68 @@ public void run() {
...
 
+    private void showError(String s) {
+        invalidCredentialsLabel.setForeground(WidgetUtils.ADDITIONAL_COLOR_RED_BRIGHT);
+        invalidCredentialsLabel.setText(s);
+    }
+
+    private void acceptTerms() {
+        try {
+            // call web service to accept T&C
+            RESTClientImpl client = new RESTClientImpl(usernameTextField.getText(), new String(passwordTextField.getPassword()));
```
Comment (ClaudiaPHI) [inline desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java]: final

File: desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java [hunk 3]
```diff
@@ -219,6 +238,53 @@ private DCPanel contentLoginCard() {
...
+        final JLabel description = new JLabel("Please review terms and conditions before using DataCloud services");
+        final JButton acceptButton = WidgetFactory.createPrimaryButton("Accept", IconUtils.ACTION_SAVE_BRIGHT);
+        final JXEditorPane tacArea = new JXEditorPane();
+        try {
+            String tac = RemoteServersUtils.getDataCloudTermsAndConditions();
+            tacArea.setContentType("text/html");
+            tacArea.setText(tac);
+            tacArea.setCaretPosition(0);
+        } catch (IOException e) {
+            // TODO
```
Comment (rposkocil) [inline desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java]: What is missing here? And what about the previous TODO? Could it be finished?

**Comment by ClaudiaPHI** [comment]
LGTM!

**Pull request merged**